app/models/katello/rhel_lifecycle_status.rb

```diff
@@ -69,7 +69,15 @@ def self.status_map
     end
 
     def self.approaching_end_of_category(eos_schedule_index:)
-      RHEL_EOS_SCHEDULE[eos_schedule_index].select { |_k, v| (Time.now.utc..Time.now.utc + EOS_WARNING_THRESHOLD).cover?(v) }
+      lifecycles_expire_soon[eos_schedule_index]
+    end
+
+    def self.lifecycles_expire_soon
+      expiring = RHEL_EOS_SCHEDULE.collect do |index, schedules|
+        expire_soon = schedules.except("full_support").select { |_k, v| (Time.now.utc..Time.now.utc + EOS_WARNING_THRESHOLD).cover?(v) }
+        {index => expire_soon} if expire_soon.present?
+      end
+      expiring.compact.reduce(:update) || {}
```

Comment:
This is confusing to me. Where is the `:update` method being passed to `reduce`? Can this be made any more clear? or maybe add some comments?

Reply:
That command is changing `[{"RHEL7"=>{"maintenance_support"=>2024-06-30 23:59:59.999999999 UTC}}, {"RHEL6"=>{"extended_support"=>2024-06-30 23:59:59.999999999 UTC}}]` 
to `{"RHEL7"=>{"maintenance_support"=>2024-06-30 23:59:59.999999999 UTC}, "RHEL6"=>{"extended_support"=>2024-06-30 23:59:59.999999999 UTC}}`.

[.update](https://ruby-doc.org/2.7.8/Hash.html#method-i-update) is an aliase of [Hash#merge!](https://ruby-doc.org/2.7.8/Hash.html#method-i-merge-21). So there is no temporary variables created during the iteration.

Reply:
Thanks for explaining :+1: 

I would prefer `:merge` so future me doesn't confuse it with Rails `update` :smile:

Reply:
I bet you have learned it. `:merge` would create a new temporary variable for each iteration that is useless.

Reply:
oh, I thought you were saying they were aliases meaning the same code. Ok then!